Sources/FirebaseFirestore/DocumentReference.swift

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Hmm, these are still Swift extensions to the C++ type right?  I think `DocumentReference+Swift.swift` might be nicer.